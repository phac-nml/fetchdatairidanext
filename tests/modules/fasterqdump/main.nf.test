nextflow_process {
    name "Test Process SRATOOLS_FASTERQDUMP"
    script "modules/local/sratools/fasterqdump/main.nf"
    process "SRATOOLS_FASTERQDUMP"
    tag "modules"
    tag "modules_nfcore"
    tag "sratools/fasterqdump"

    test("Test fasterqdump ERR1109373") {

        when {
            process {
                """
                input[0] = Channel.of([ [irida_id: "SAMPLE1"], file("$baseDir/tests/data/ERR1109373.sra", checkIfExists: true) ])
                input[1] = file('$baseDir/tests/data/ncbi_user_settings.mkfg', checkIfExists: true)
                input[2] = []
                """
            }

            params {
                outdir = "fasterqdump_test1_out"
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
