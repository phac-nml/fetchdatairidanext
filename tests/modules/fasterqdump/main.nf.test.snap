{
    "Test fasterqdump ERR1109373": {
        "content": [
            {
                "0": [
                    [
                        {
                            "irida_id": "SAMPLE1"
                        },
                        [
                            "ERR1109373.fastq.gz:md5,b9acccb3c5d317a99f604375a09991aa",
                            "ERR1109373_1.fastq.gz:md5,38e94cf16bf3d3b7d3a05a4c21b8fcc3",
                            "ERR1109373_2.fastq.gz:md5,031a82c07f35c45ada5b6133b1181e68"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,6ff2d50b15c3f0eb9c72cd13a4a20295"
                ],
                "reads": [
                    [
                        {
                            "irida_id": "SAMPLE1"
                        },
                        [
                            "ERR1109373.fastq.gz:md5,b9acccb3c5d317a99f604375a09991aa",
                            "ERR1109373_1.fastq.gz:md5,38e94cf16bf3d3b7d3a05a4c21b8fcc3",
                            "ERR1109373_2.fastq.gz:md5,031a82c07f35c45ada5b6133b1181e68"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6ff2d50b15c3f0eb9c72cd13a4a20295"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-11T20:36:55.563890217"
    }
}